### PR TITLE
Test: Add timeout handling test for DataStorage read operation

### PR DESCRIPTION
**Pull request by mirahsupreme**
this helps if anything like delays occur and can make the DataStorage more stable and precise.

### CODE REVIEW

File: test/TestDataStorageTimeout.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        ReadRequest mockRequest = Mockito.mock(ReadRequest.class);
+
+        try {
+            // Configure the mock to throw a TimeoutException for the read method
+            Mockito.when(ds.read(mockRequest)).thenThrow(new TimeoutException("Read operation timed out"));
+
+            // Call the read method, expecting a TimeoutException
+            ds.read(mockRequest);
+            fail("Expected a TimeoutException to be thrown");
+        } catch (TimeoutException e) {
```
Comment (Oliver-222) [inline test/TestDataStorageTimeout.java]: exception TimeoutException is never thrown in body of corresponding try statement

Comment (mirahsupreme) [reply test/TestDataStorageTimeout.java]: Okay it's fixed!

**Comment by mirahsupreme** [comment]
can you guys approve this please

**Review by loganullmeyer — APPROVED** [review]
Fixed, all checks passing, LGTM

**Pull request merged**